chain/client/src/chunk_inclusion_tracker.rs

```diff
@@ -0,0 +1,94 @@
+use itertools::Itertools;
+use lru::LruCache;
+use std::collections::HashMap;
+
+use near_primitives::hash::CryptoHash;
+use near_primitives::sharding::ShardChunkHeader;
+use near_primitives::types::{AccountId, EpochId, ShardId};
+
+use crate::metrics;
+
+const CHUNK_HEADERS_FOR_INCLUSION_CACHE_SIZE: usize = 2048;
+const NUM_EPOCH_CHUNK_PRODUCERS_TO_KEEP_IN_BLOCKLIST: usize = 1000;
+
+pub struct ChunkInclusionTracker {
+    prev_block_to_chunk_headers_ready_for_inclusion: LruCache<
```

Comment:
As you are moving it, it can be beneficial to leave comment what this scary thing does :) And leave comments for mark_chunk_header_ready_for_inclusion and get_chunk_headers_ready_for_inclusion